server/controllers/plugins.js

```diff
@@ -0,0 +1,99 @@
+const asyncHandler = require("express-async-handler");
+const Card = require("./../models/Card");
+const BasePlugin = require("../models/BasePlugin");
+
+exports.activate = asyncHandler(async (req, res) => {
+  const { pluginName } = req.params;
+  const response = await BasePlugin.attach(Card, pluginName);
+
+  if (response.status === 400) {
+    res.status(400).json({ message: response.message });
+  }
+
+  res.status(200).json({ response });
+});
+
+/**
+ * Removes a plugin from a card and destroys it.
+ * @function
+ */
+exports.destroy = asyncHandler(async (req, res) => {
+  const { pluginId } = req.params;
+  const plugin = await BasePlugin.detach(Card, pluginId);
+
+  if (!plugin) {
+    res.status(400).json({ message: "Plugin does not exist." });
+  }
+
+  const deletedPlugin = await BasePlugin.findByIdAndDelete(pluginId);
+  res.status(200).json({ plugin: deletedPlugin });
+});
+
+/**
+ * Retrieves all the plugins in a particular card.
+ * @function
+ */
+exports.getAllPlugins = asyncHandler(async (req, res) => {
+  const { cardId } = req.params;
+  const card = await Card.findById(cardId);
+
+  if (!card) {
+    res.status(400).json({ message: "Card does not exist." });
+  }
+
+  res.status(200).json({ plugins: card.plugins });
+});
+
+/**
+ * Retrieves a plugin.
+ * @function
+ */
+exports.getPlugin = asyncHandler(async (req, res) => {
+  const response = await req.plugin.get();
+  if (response.status === 200) {
+    res.status(200).json(response.data);
+  } else {
+    res.status(400).json({ message: "Plugin not available" });
+  }
```

Comment:
I think my recommendation above would be better. `res.status(response.status).json(response.data)` 

It will put the responses on the plugin.